servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/CoreMessageReceiver.java

```diff
@@ -1176,14 +1167,7 @@ public void onEvent() {
             CoreMessageReceiver.this.ensureLinkIsOpen().thenRun(() -> {
                 try
                 {
-                    this.underlyingFactory.scheduleOnReactorThread(new DispatchHandler()
-                    {
-                        @Override
-                        public void onEvent()
-                        {
-                            delivery.disposition(state);
-                        }
-                    });
+                    this.underlyingFactory.scheduleOnReactorThread(new DeliveryStateDispatchHandler(delivery, state));
```

Comment:
[SpotBugs-P2]
This class is an inner class, but does not use its embedded reference to the object which created it.  This reference makes the instances of the class larger, and may keep the reference to the creator object alive longer than necessary.  If possible, the class should be made static.